src/hdx/scraper/utilities/reader.py

```diff
@@ -317,6 +318,16 @@ def read_hdx_metadata(
         dataset_nameinfo = datasetinfo["dataset"]
         if isinstance(dataset_nameinfo, str):
             dataset = self.read_dataset(dataset_nameinfo)
```

Comment:
the code in this if statement (since the addition of the hapi metadata part) is pretty long, maybe it warrants it's own method?

Reply:
I created a couple of methods to deal with the HAPI dataset and resource